tests/console/validate_zfcp.pm

```diff
@@ -0,0 +1,199 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate zfcp in the installed system:
+#          - Verification of FCP devices online
+#          - Verificaiton of FCP devices
+#          - Verification of LUNs visible as SCSI devices
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use utils qw(arrays_subset);
+
+# Get devices online.
+# For each physical adapter in the mainframe there is a corresponding FCP channel providing physical
+# connection to the SAN Fabric (multiple FCP channels are configured to increase the I/O bandwidth
+# and improve data availability).
+# HBAs are shown in the system as FCP devices
+sub get_fcp_devices_online {
+    my $test_data = get_test_suite_data();
+
+    # FCP devices (online and offline)
+    my $fcp_devices = $test_data->{zfcp}->{fcp_devices};
+    # filtering online devices
+    return [grep { $_->{attributes}->{online} } @{$fcp_devices}];
+}
+
+# Compare fcp devices
+sub verify_fcp_devices {
+    my (%args) = @_;
+
+    my $devs_expected = [map { $_->{fcp_channel} } @{$args{devs_expected}}];
+    my $devs_current  = $args{devs_current};
+    my $only_warning  = $args{only_warning};
+
+    my @dif = arrays_subset($devs_expected, $devs_current);
+    if (scalar @dif > 0) {
+        my $message = "Unexpected FCP devices online found in the installed system. " .
+          "Please, check if it was an error or is intended and new adaptor became renamed " .
```

Comment:
i think you meant it without the `became`. `and new adaptor renamed ", right?

Reply:
looks fine to me, or (1) it is an error or (2) it is intended and new adaptor became renamed. Perhaps is missing some 'it' but I think is readable enough.

Reply:
no idea here. ignore it if you want